packages/babel-plugin-transform-es2015-object-super/src/index.js

```diff
@@ -1,46 +1,39 @@
 import ReplaceSupers from "babel-helper-replace-supers";
 
-export default function ({ types: t }) {
-  function Property(path, node, scope, getObjectRef, file) {
-    const replaceSupers = new ReplaceSupers({
-      getObjectRef: getObjectRef,
-      methodNode: node,
-      methodPath: path,
-      isStatic: true,
-      scope: scope,
-      file: file,
-    });
-
-    replaceSupers.replace();
-  }
-
-  const CONTAINS_SUPER = Symbol();
+function replacePropertySuper(path, node, scope, getObjectRef, file) {
+  const replaceSupers = new ReplaceSupers({
+    getObjectRef: getObjectRef,
+    methodNode: node,
+    methodPath: path,
+    isStatic: true,
+    scope: scope,
+    file: file,
+  });
+
+  replaceSupers.replace();
+}
 
+export default function ({ types: t }) {
   return {
     visitor: {
-      Super(path) {
-        const parentObj = path.findParent((path) => path.isObjectExpression());
-        if (parentObj) parentObj.node[CONTAINS_SUPER] = true;
-      },
+      ObjectExpression(path, state) {
+        let objectRef;
+        const getObjectRef = () => objectRef = objectRef || path.scope.generateUidIdentifier("obj");
 
-      ObjectExpression: {
-        exit(path, file) {
```

Comment:
This implementation had to change because the usage of the `exit` handler meant that the arrow transform would fail for
```
var obj = {
  method(){
    var fn = () => super.method();
  }
};
```
since it can't handle `super`, meaning that for object-super to work, it needs to run before the arrow transform.